### DISCUSSION TITLE
Improving 'first experience' with LXQt on Arch e.g.

### DISCUSSION THREAD

**Discussion by Frtzqe** [lxqt/lxqt#2661] [General]
**LXQt** is not very well known to the general public, despite that it is rather a nice desktop environment. The best DE I can imagine. Unfortunately it seems that LXQt has ended up in some unknown alley, _if_ provided by some distribution. and, if nothing happens, it will remain a _niche_ DE. 

Paraphrasing the Cambridge dictionary of English, the meaning of _niche_, with respect to LXQt, is: 

`a desktop environment that is exactly suitable for a small group of the same type`

I am not Nostradamus and surely I did not see all the current various offerings of **_a LXQt implementation_** in the swamp of package repositories out there (_group_, _meta-package_, etc, what ever they call it).

There are several excellent examples: Lubuntu, EndeavourOS, Garuda, etc. 

But, for people like me, who like to stay as close as possible to a clean Linux system, and others who do not know how simple it is to customise LXQt,  it is important that LXQt has a decent **"First Impression"** implementation. 

In particular with **Linux Arch** the first experience with LXQt after _pacman_-ing their _lxqt_ group is _not as nice_ as should be. With respect for the maintainer(s) of the said group.....  Novice candidate users of LXQt might give up after 2, 3 minutes and never experience the true beauty of LXQt. 

No modern looking theme, no icons implemented. It really looks (without some simple customisations) even worse than a 1990's X11 desktop with just Motif on a Sun workstation.

So, how are we to help the various distribution maintainers to just showcase LXQt out of the box?

NOTE: I cleaned up the original text a bit; my original qualifications unpolished, unfinished and appalling were a bit _crude_.

**Comment by stefonarch** [comment]
An "old" issue, see https://github.com/lxqt/lxqt/discussions/2163
This said XFCE defaults are real ugly too on Arch. 
For the missing icons maybe we could add a dependency to the set, the default config looks for breeze: https://github.com/lxqt/lxqt-session/blob/master/config/lxqt.conf

**Comment by Frtzqe** [comment]
I don't care too much about XFCE (in fact, I do, but this is the LXQt Discussion site, right?), so...

My bet would be to encourage the LXQt developer(s) to do their best a bit, and add a nice default. Now **their defaul is very ugly**.

I think they are proud about LXQt being agnostic in any way, but _**they can do better**_.

**Reply by Frtzqe** [reply]
I am starting to wonder who is administering the `lxqt` group on Arch?

**Reply by isf63** [reply]
Arch Linux by design is a "do-it-yourself" distro. A problem it has with fresh LXQt installations is that there is no `Provides: icon-theme` standard (like the existing `Provides: ttf-font`. (Edit: This is a problem because LXQt is icon theme agnostic.) One possibility at least is adding `Depends On: ttf-font` to every LXQt package - Qt on Arch doesn't pull it in.

**Reply by tsujan** [reply]
> Arch Linux by design is a "do-it-yourself" distro.

OK. But Arch packages have an "Optional Deps" section. My question: Is an appropriate icon set an optional dependency of an LXQt package? If the answer is "no", Arch has a problem to fix; if it's "yes", it's *the user* that should care for optional dependencies and install them. In the second case, there's no problem to solve.

**Reply by isf63** [reply]
Without something like `Depends: icon-theme` there'd need to be one or more specific icon themes in optdeps. This I think is no good. Also, it is rare for GUI user applications to require `ttf-font` on Arch, on my system for ttf-font: `Required By     : firefox  sddm  steam  thunderbird`. So yeah, not much a problem to solve.

**Reply by tsujan** [reply]
> This I think is no good.

Why? It's optional, after all.

**Reply by tsujan** [reply]
This is what Debian (Alf) did: https://packages.debian.org/buster/lxqt-system-theme

Debian can use "Or" for its "Recommends" (the counterpart of "Optional Deps" in Arch). I don't know if Arch can have "Or" or not, but at least one icon theme is needed.

**Reply by tsujan** [reply]
@yan12125
Any thought on this?

**Reply by yan12125** [reply]
I haven't read the entire thread. First batch of my thoughts:

> For the missing icons maybe we could add a dependency to the set, the default config looks for breeze: https://github.com/lxqt/lxqt-session/blob/master/config/lxqt.conf

> Is an appropriate icon set an optional dependency of an LXQt package?

Usually Arch follows upstream decisions. If upstream declares a preferred/default icon theme, Arch packages can have it in optional dependencies.

> One possibility at least is adding `Depends On: ttf-font` to every LXQt package

The direction is good. I prefer to do it for only lxqt-session or a few selected packages instead of everyone.

> I don't know if Arch can have "Or"

Well, its package manager `pacman` does not support that.

**Reply by tsujan** [reply]
> If upstream declares a preferred/default icon theme, Arch packages can have it in optional dependencies.

IMHO, Breeze is a good optional dependency. @stefonarch also seems to suggest it with that comment. And you're in both upstream and downstream :)

**Reply by yan12125** [reply]
I added ttf-font. On the other hand, `breeze-icons` was added as an optional dependency long time ago: https://gitlab.archlinux.org/archlinux/packaging/packages/lxqt-session/-/commit/ea4c83cb86344e1a022994c27f68a6f8cc4b47d6. Maybe that's a sign that an optional dependency is not enough.

**Comment by Frtzqe** [comment]
As a start, a default, when you just _pacman_ `lxqt` and get something like https://lxqt-project.org/images/screenshots/ambiance.png would be really great, IMHO. 

I emphasise: without tweaking anything :-)  Just out the box. 

How hard can that be?? Perhaps even with PCManFM-qt and Qterminal already available as quicklaunch shortcut and **xfwm4** as window manager :-)

**Reply by stefonarch** [reply]
Without tweaking anything icon theme is breeze and background/theme Clearlooks, which is was arch is shipping (except breeze by default). That was discussed and could be changed (where we have as many tasts as partecipants) but it will affect anyway only Arch users which are expect to tune things themselves.

![screen](https://github.com/user-attachments/assets/ff7f0aed-d21c-444b-9f0c-ebb417ce619a)

The WM manager has to be chosen by the user, if more than one is installed. 
Fine tuning everything from theme to icons to font is a distro job. Lubuntu ships openbox, debian xfwm4.

**Reply by Frtzqe** [reply]
I agree with you that Arch users may be stubborn. I personally think that is a good thing :-) 

But, although simple, even LXQt has a learning curve, and when you "occasionally" install the `lxqt` group on _Arch Linux_ (curious, without having any knowledge about LXQt), the result is very unpolished and unfinished. I will try to send a desktop-shot later.

**Comment by tsujan** [comment]
I'd like to add something obvious that's often ignored in such discussions: one man's meat is another man's poison.

"Appalling", "unpolished", "unfinished",… are words that refer to tastes.

**Reply by Frtzqe** [reply]
Yeah, I know you put a lot of effort (and others too) into LXQt. 

I can't emphasise enough how much I appreciate that. I also understand that what I am talking about is a _distro thingy_.

But, really, 

![screen_lxqt_on_arch](https://github.com/user-attachments/assets/e07464d1-5a35-4d4c-aea5-2750645d0877)

...is this how you would like to promote your LXQt to novice users on Arch Linux? 

I hope not. Quoting my own question at the end of the start of this discussion: 

`So, how are we to help the various distribution maintainers to just showcase LXQt out of the box?`

In this state, LXQt is unfinished, unpolished, and even appalling, Clearly not how you want to showcase it. 
Right? Not exactly like the _Ambiance_ picture shown prominently on the lxqt-project website.

For your information: all the novice user on Arch Linux did was:
` $ sudo pacman -S sddm lxqt; sudo systemctl enable --now sddm`

**Reply by tsujan** [reply]
> But, really,

LOL! That's not unfinished or unpolished; it's wrong ;) And, as far as your screenshot shows, it's only about the icon theme. The title of your discussion could have been: No icons with Arch's LXQt.

**Reply by tsujan** [reply]
Also, see https://github.com/lxqt/lxqt/discussions/2661#discussioncomment-11605412

**Reply by Frtzqe** [reply]
Sure, there is no accounting for taste. And, yes, that the icon theme is missing is _killing_.

Even with a proper icon theme, the default Arch Linux delivery is _ugly_ IMHO. So, a task for the lxqt-project team to provide a nice modern default, or is this the task for each distribution? 

If the project doesn't, I am probably barking at the wrong tree here. And should change my location to the maintainer(s) of the Arch Linux `lxqt` group. 

Sounds reasonable, but I would prefer that the project takes its responsibility.

**Reply by tsujan** [reply]
Your main point, which is about icons, is quite valid. Arch should fix it.

Whether the default LXQt theme is ugly or not is subjective though. This was what I thought you were talking about; hence my first reply. The user can select another LXQt theme, change the WM to a fancy one like KWin — with lots of effects — use a widget style like Kvantum instead of Fusion, and so on. LXQt is agnostic toward these. Particularly in Arch, they should be the user's choices.

But an appropriate icon set as an *optional* dependency is Arch's job.

**Comment by tsujan** [comment]
IMHO, Arch's [lxqt-themes](https://archlinux.org/packages/extra/any/lxqt-themes/) should *optionally* depend on a good icon theme. `hicolor-icon-theme` is good as a usual dependency but absolutely not enough.

That should be fixed in Arch.

**Reply by stefonarch** [reply]
It's simple: if breeze icons aren't installed there will be no icons. As we set it in the default config file we have 2 options:

1. Make it a dependency
2. Use a mechanism for a fallback icon set if breeze isn't present.

**Reply by tsujan** [reply]
> we have 2 options

We have nothing to solve, IMO.

**Reply by Frtzqe** [reply]
Very good idea. A certain fallback will avoid _icon-less_ implementations (in the future). 

Further, it should be up to the distro maintainer to offer a few "go with the flow" alternatives, like `papirus-icon-theme`. 

Why should I not have a _GwtF_ icon set or theme selection when I install LXQt (e.g. some Arc variant), but _au contrair_ have a plethora of choices for the font in `sddm` when I install it? Last sentence is too long. Nobody should care too much about the specific font in `sddm`, but having a choice for icon set and/or theme while `pacman`-ing `lxqt` would be really great.

Same goes for the window manager. Why not let the user decide if `openbox` should be used, or e.g. `xfwm4` (which would be an excellent choice)?

I really need to talk to the Arch Linux guy in charge of the LXQt group packaging.

And yeah....Archers should know how to tune their desktop. However, "first impressions" count :-)

**Reply by Frtzqe** [reply]
> > we have 2 options
> 
> We have nothing to solve, IMO.

I dont' like this attitude (response). I've demonstrated a certain issue. Neglecting it is too easy.

**Reply by tsujan** [reply]
It isn't related to likes or dislikes. Please read above again.

**Reply by yan12125** [reply]
> IMHO, Arch's [lxqt-themes](https://archlinux.org/packages/extra/any/lxqt-themes/) should optionally depend on a good icon theme.

As mentioned above, lxqt-session already optionally depends on breeze-icons since long time ago. I can add it, but that will likely solve nothing.

> I really need to talk to the Arch Linux guy in charge of the LXQt group packaging.

Here am I. However, due to technical limitations, something like `Provides: icon-theme` (similar to `Provides: ttf-font`) seems the only viable way. That involves various changes in non-LXQt packages. I should have time to write a proposal for Arch Linux during the weekend.

**Reply by tsujan** [reply]
> As mentioned above, lxqt-session already optionally depends on breeze-icons since long time ago.

Oh, I see that lxqt-session optionally depends on both Breeze and Oxygen. Sorry for the fuss! I should have searched more Arch packages.

@Frtzqe
You missed the optional dependencies. They don't exist just for fun.

**Reply by isf63** [reply]
@yan12125,

`Provides: ttf-font` is used on Arch, but `Depends On: ttf-font` is hardly used. I predict a `Provides: icon-theme` would end up in the same boat.

**Reply by Frtzqe** [reply]
> They don't exist just for fun.

Yeah, you are totally right. Thanks for pointing that out to me. 

From the Arch wiki:  
> First, install and configure Xorg. Then, install the lxqt group **and** an icon theme (e.g. breeze-icons or oxygen-icons).

The emphasise is by me :-)  You know how things work, when there is a "group" that could be used to get LXQt up and running, one may have certain expectations.  People tend to believe....well..."That should do it", and get dissatisfied with the result _so far_. 

I know how to customise LXQt. But not every newbie who got fixed with Gnome or Plasma, and wants to try something else (for various reasons), does know how to do this quickly and easily! 

Btw: I really like artwork done by the Lubuntu team when they moved from LXDE to LXQt. With a few strokes at the keyboard *any* LXQt implementation could look and feel almost exactly the same like Lubuntu. Than you Lubuntu team!

**Reply by yan12125** [reply]
> Provides: ttf-font is used on Arch, but Depends On: ttf-font is hardly used. I predict a Provides: icon-theme would end up in the same boat.

As per our goal, it's enough to have only one package that uses `Depends On: icon-theme` - that is, lxqt-session.

**Reply by stefonarch** [reply]
OT: https://aur.archlinux.org/packages/lxqt-wayland-session-git could be removed now, no?

**Reply by yan12125** [reply]
> OT: https://aur.archlinux.org/packages/lxqt-wayland-session-git could be removed now, no?

I prefer to keep it just like other lxqt -git packages on AUR.

**Reply by stefonarch** [reply]
Oops, sure, it makes sense as it can be ahead.

**Comment by Frtzqe** [comment]
Thank you all for your constructive reactions. Appreciated. Btw I am a big fan of LXQt (earlier LXDE already), and trying to promote it where and whenever I can.

**Comment by Frtzqe** [comment]
Because people are reading this, and as a personal reference, the following gives a good technical introduction to what is this all about (theming that is):

https://github.com/lxqt/lxqt/wiki/Creating-LXQt-Themes   

Tsujan is author of a general Qt theming engine, which might be useful as well: 

https://github.com/tsujan/Kvantum
